### ISSUE TITLE
Build fails for evm using r2pm

### ISSUE THREAD

**Issue by chhajershrenik** [radareorg/radare2-extras#323]
### Environment

```
radare2 -V
5.8.2-11-gabb7a7c749  r2
5.8.2-11-gabb7a7c749  r_anal
5.8.2-11-gabb7a7c749  r_lib
5.8.2-11-gabb7a7c749  r_egg
5.8.2-11-gabb7a7c749  r_asm
5.8.2-11-gabb7a7c749  r_bin
5.8.2-11-gabb7a7c749  r_cons
5.8.2-11-gabb7a7c749  r_flag
5.8.2-11-gabb7a7c749  r_core
5.8.2-11-gabb7a7c749  r_crypto
5.8.2-11-gabb7a7c749  r_bp
5.8.2-11-gabb7a7c749  r_debug
5.8.2-11-gabb7a7c749  r_main
5.8.2-11-gabb7a7c749  r_fs
5.8.2-11-gabb7a7c749  r_io
5.8.2-11-gabb7a7c749  r_magic
5.8.2-11-gabb7a7c749  r_parse
5.8.2-11-gabb7a7c749  r_reg
5.8.2-11-gabb7a7c749  r_sign
5.8.2-11-gabb7a7c749  r_search
5.8.2-11-gabb7a7c749  r_syscall
5.8.2-11-gabb7a7c749  r_util

date
Sun Jan 29 09:08:11 AM EST 2023

r2 -v
radare2 5.8.3 29841 @ linux-x86-64 git.5.8.2-11-gabb7a7c749
commit: abb7a7c749ca128b46426b05e2efaa138e254041 build: 2023-01-29__05:23:17

uname -ms
Linux x86_64
```

### Description

r2pm build fails for evm.

### Test
```bash
r2pm -ci evm
```
```bash
INFO: Cleaning evm
INFO: Using r2-5.8.3 and r2pm-5.8.3
INFO: Cleaning evm
INFO: git clone --depth=10 --recursive https://github.com/radareorg/radare2-extras /home/kali/.local/share/radare2/r2pm/git//evm
Cloning into '/home/kali/.local/share/radare2/r2pm/git//evm'...
remote: Enumerating objects: 1243, done.
remote: Counting objects: 100% (1243/1243), done.
remote: Compressing objects: 100% (1069/1069), done.
remote: Total 1243 (delta 144), reused 1020 (delta 102), pack-reused 0
Receiving objects: 100% (1243/1243), 6.95 MiB | 2.89 MiB/s, done.
Resolving deltas: 100% (144/144), done.
INFO: Starting install for evm
script (        ./configure --prefix="${R2PM_PREFIX}" || exit 1
        cd libr
        ${MAKE} clean
        ${MAKE} evm || exit 1
        ${MAKE} evm-install R2PM_PLUGDIR="${R2PM_PLUGDIR}")
checking build system type... x86_64-unknown-linux-gnu
checking host system type... x86_64-unknown-linux-gnu
checking target system type... x86_64-unknown-linux-gnu
checking for working directories... current
using prefix '/home/kali/.local/share//radare2/prefix'
checking for pkg-config... /usr/bin/pkg-config
checking for c compiler... gcc
Using PKGCONFIG: /usr/bin/pkg-config
checking pkg-config flags for r_core... yes
checking for libewf ... no
checking pkg-config flags for libewf... no
checking pkg-config flags for unicorn... no
checking for libyara ... no
checking pkg-config flags for yara... no
creating ./Makefile
creating ./options.mk
cleaning temporally files... done

Final report:
 - PREFIX = /home/kali/.local/share//radare2/prefix
make -C asm clean
make[1]: Entering directory '/home/kali/.local/share/radare2/r2pm/git/evm/libr/asm'
rm -f p/*.so p/*.o
make[1]: Leaving directory '/home/kali/.local/share/radare2/r2pm/git/evm/libr/asm'
make -C anal clean
make[1]: Entering directory '/home/kali/.local/share/radare2/r2pm/git/evm/libr/anal'
rm -f p/anal_ba2.so
rm -f p/anal_x86_bea.so
rm -f p/anal_m68k_net.so
rm -f p/anal_atombios.{so,o}
rm -f p/anal_nios.so
rm -f p/anal_hexagon.so
make[1]: Leaving directory '/home/kali/.local/share/radare2/r2pm/git/evm/libr/anal'
make -C asm evm
make[1]: Entering directory '/home/kali/.local/share/radare2/r2pm/git/evm/libr/asm'
make -C p asm_evm.so
make[2]: Entering directory '/home/kali/.local/share/radare2/r2pm/git/evm/libr/asm/p'
gcc -c -I/usr/local/include/libr -g -I/shlr -I../../include -I../arch/ -I../arch/include -Wall -shared -fPIC  -DLIL_ENDIAN=1 -I../arch/evm -Wall -shared -fPIC  asm_evm.c -o asm_evm.o
asm_evm.c:11:33: error: unknown type name ‘RAsmOp’; did you mean ‘RAsm’?
   11 | static int disassemble(RAsm *a, RAsmOp *op, const ut8 *buf, int len) {
      |                                 ^~~~~~
      |                                 RAsm
asm_evm.c:23:30: error: unknown type name ‘RAsmOp’; did you mean ‘RAsm’?
   23 | static int assemble(RAsm *a, RAsmOp *op, const char *buf) {
      |                              ^~~~~~
      |                              RAsm
asm_evm.c:28:1: error: unknown type name ‘RAsmPlugin’; did you mean ‘RAnalPlugin’?
   28 | RAsmPlugin r_asm_plugin_evm = {
      | ^~~~~~~~~~
      | RAnalPlugin
asm_evm.c:29:9: error: field name not in record or union initializer
   29 |         .name = "evm",
      |         ^
asm_evm.c:29:9: note: (near initialization for ‘r_asm_plugin_evm’)
asm_evm.c:29:17: warning: initialization of ‘int’ from ‘char *’ makes integer from pointer without a cast [-Wint-conversion]
   29 |         .name = "evm",
      |                 ^~~~~
asm_evm.c:29:17: note: (near initialization for ‘r_asm_plugin_evm’)
asm_evm.c:29:17: error: initializer element is not computable at load time
asm_evm.c:29:17: note: (near initialization for ‘r_asm_plugin_evm’)
asm_evm.c:30:9: error: field name not in record or union initializer
   30 |         .author = "pancake",
      |         ^
asm_evm.c:30:9: note: (near initialization for ‘r_asm_plugin_evm’)
asm_evm.c:30:19: warning: excess elements in scalar initializer
   30 |         .author = "pancake",
      |                   ^~~~~~~~~
asm_evm.c:30:19: note: (near initialization for ‘r_asm_plugin_evm’)
asm_evm.c:31:9: error: field name not in record or union initializer
   31 |         .version = "0.0.1",
      |         ^
asm_evm.c:31:9: note: (near initialization for ‘r_asm_plugin_evm’)
asm_evm.c:31:20: warning: excess elements in scalar initializer
   31 |         .version = "0.0.1",
      |                    ^~~~~~~
asm_evm.c:31:20: note: (near initialization for ‘r_asm_plugin_evm’)
asm_evm.c:32:9: error: field name not in record or union initializer
   32 |         .arch = "evm",
      |         ^
asm_evm.c:32:9: note: (near initialization for ‘r_asm_plugin_evm’)
asm_evm.c:32:17: warning: excess elements in scalar initializer
   32 |         .arch = "evm",
      |                 ^~~~~
asm_evm.c:32:17: note: (near initialization for ‘r_asm_plugin_evm’)
asm_evm.c:33:9: error: field name not in record or union initializer
   33 |         .license = "MIT",
      |         ^
asm_evm.c:33:9: note: (near initialization for ‘r_asm_plugin_evm’)
asm_evm.c:33:20: warning: excess elements in scalar initializer
   33 |         .license = "MIT",
      |                    ^~~~~
asm_evm.c:33:20: note: (near initialization for ‘r_asm_plugin_evm’)
asm_evm.c:34:9: error: field name not in record or union initializer
   34 |         .bits = 32,
      |         ^
asm_evm.c:34:9: note: (near initialization for ‘r_asm_plugin_evm’)
asm_evm.c:34:17: warning: excess elements in scalar initializer
   34 |         .bits = 32,
      |                 ^~
asm_evm.c:34:17: note: (near initialization for ‘r_asm_plugin_evm’)
asm_evm.c:35:9: error: field name not in record or union initializer
   35 |         .endian = R_SYS_ENDIAN_BIG,
      |         ^
asm_evm.c:35:9: note: (near initialization for ‘r_asm_plugin_evm’)
In file included from asm_evm.c:5:
/usr/local/include/libr/r_types.h:621:26: warning: excess elements in scalar initializer
  621 | #define R_SYS_ENDIAN_BIG 2
      |                          ^
asm_evm.c:35:19: note: in expansion of macro ‘R_SYS_ENDIAN_BIG’
   35 |         .endian = R_SYS_ENDIAN_BIG,
      |                   ^~~~~~~~~~~~~~~~
/usr/local/include/libr/r_types.h:621:26: note: (near initialization for ‘r_asm_plugin_evm’)
  621 | #define R_SYS_ENDIAN_BIG 2
      |                          ^
asm_evm.c:35:19: note: in expansion of macro ‘R_SYS_ENDIAN_BIG’
   35 |         .endian = R_SYS_ENDIAN_BIG,
      |                   ^~~~~~~~~~~~~~~~
asm_evm.c:36:9: error: field name not in record or union initializer
   36 |         .desc = "evm",
      |         ^
asm_evm.c:36:9: note: (near initialization for ‘r_asm_plugin_evm’)
asm_evm.c:36:17: warning: excess elements in scalar initializer
   36 |         .desc = "evm",
      |                 ^~~~~
asm_evm.c:36:17: note: (near initialization for ‘r_asm_plugin_evm’)
asm_evm.c:37:9: error: field name not in record or union initializer
   37 |         .disassemble = &disassemble,
      |         ^
asm_evm.c:37:9: note: (near initialization for ‘r_asm_plugin_evm’)
asm_evm.c:37:25: error: ‘disassemble’ undeclared here (not in a function)
   37 |         .disassemble = &disassemble,
      |                         ^~~~~~~~~~~
asm_evm.c:37:24: warning: excess elements in scalar initializer
   37 |         .disassemble = &disassemble,
      |                        ^
asm_evm.c:37:24: note: (near initialization for ‘r_asm_plugin_evm’)
asm_evm.c:38:9: error: field name not in record or union initializer
   38 |         .assemble = &assemble
      |         ^
asm_evm.c:38:9: note: (near initialization for ‘r_asm_plugin_evm’)
asm_evm.c:38:22: error: ‘assemble’ undeclared here (not in a function)
   38 |         .assemble = &assemble
      |                      ^~~~~~~~
asm_evm.c:38:21: warning: excess elements in scalar initializer
   38 |         .assemble = &assemble
      |                     ^
asm_evm.c:38:21: note: (near initialization for ‘r_asm_plugin_evm’)
make[2]: *** [evm.mk:14: asm_evm.o] Error 1
make[2]: Leaving directory '/home/kali/.local/share/radare2/r2pm/git/evm/libr/asm/p'
make[1]: *** [Makefile:43: evm] Error 2
make[1]: Leaving directory '/home/kali/.local/share/radare2/r2pm/git/evm/libr/asm'
make: *** [Makefile:37: evm] Error 2
````

**Comment by trufae** [comment]
Evm support is shipped in r2 no need to build this plugin from extras. R2-5.8 deprecated the asm plugins so its expected not to work.

**Comment by chhajershrenik** [comment]
@trufae Thanks for the update, I was building it from extras because the opcodes were not translated into instructions when analyzing a binary.

Was getting the following errors when using the latest build for radare2 from master branch.
```bash
ERROR: Cannot find 'evm' asm/arch/anal plugin. See rasm2 -L or -LL
ERROR: Cannot find 'evm' asm/arch/anal plugin. See rasm2 -L or -LL
```

**Comment by trufae** [comment]
theh plugin name in core is the capstone one, so the name is `evm.cs` . but i think that should be renamed to `evm` because it should be the default. let me do that change in a PR

**Comment by chhajershrenik** [comment]
Thank you. Works great if specified evm.cs as arch. 
Can close this issue :)

**Comment by trufae** [comment]
ill merge the evm.cs code and the one here and make a single plugin that will be available in r2-5.8.4 without the need of this. because the one in core miesses the assembler and there's some lack of information like esil or instruction descriptions. wanna help on this?

**Comment by chhajershrenik** [comment]
@trufae Sure happy to help, also I believe there is a room for improvement function identification and function name signature translation (https://www.4byte.directory/).
